Sources/Interpreter/Interpreter.swift

```diff
@@ -10,48 +10,32 @@ struct CodePointer {
 
 }
 
+/// The value produced by executing an instruction.
+typealias InstructionResult = Any
+
+/// Address to allocated object with type layout.
```

Comment:
Objects aren't allocated; memory is.  And I don't think there's necessarily an object there, which would require an initialization to have been performed.

This is the best I could  come up with:

```suggestion
/// A typed location in memory.
```